smartsim/_core/launcher/dragon/dragonConnector.py

```diff
@@ -0,0 +1,454 @@
+                f"Successful handshake with Dragon server at address {address}"
+            )
+        except (zmq.ZMQError, zmq.Again) as e:
+            logger.debug(e)
+            self._dragon_head_socket.close()
+            self._dragon_head_socket = None
+
+            raise SmartSimError(
+                f"Unsuccessful handshake with Dragon server at address {address}"
+            ) from e
+
+    def _reset_timeout(self, timeout: int = get_config().dragon_server_timeout) -> None:
+        self._context.setsockopt(zmq.SNDTIMEO, value=timeout)
+        self._context.setsockopt(zmq.RCVTIMEO, value=timeout)
+        if self._authenticator is not None and self._authenticator.thread is not None:
+            try:
+                self._authenticator.thread.authenticator.zap_socket.setsockopt(
+                    zmq.SNDTIMEO, timeout
+                )
+                self._authenticator.thread.authenticator.zap_socket.setsockopt(
+                    zmq.RCVTIMEO, timeout
+                )
+            except zmq.ZMQError:
+                pass
+
+    def ensure_connected(self) -> None:
+        if not self.is_connected:
+            self.connect_to_dragon()
+        if not self.is_connected:
+            raise SmartSimError("Could not connect to Dragon server")
+
+    def _get_new_authenticator(
+        self, timeout: int = get_config().dragon_server_timeout
+    ) -> None:
+        if self._authenticator is not None:
+            if self._authenticator.thread is not None:
+                try:
+                    logger.debug("Closing ZAP socket")
+                    self._authenticator.thread.authenticator.zap_socket.close()
+                except Exception as e:
+                    logger.debug(f"Could not close ZAP socket, {e}")
+            try:
+                self._authenticator.stop()
+            except zmq.Again:
+                logger.debug("Could not stop authenticator")
+        try:
+            self._authenticator = dragonSockets.get_authenticator(
+                self._context, timeout
+            )
+            return
+        except RuntimeError as e:
+            logger.error("Could not get authenticator")
+            raise e from None
+
+    @staticmethod
+    def _get_dragon_log_level() -> str:
+        smartsim_to_dragon = {
+            "developer": "INFO",
+            "debug": "NONE",
+            "info": "NONE",
+            "quiet": "NONE",
+        }
+        return smartsim_to_dragon.get(get_config().log_level, "NONE")
+
+    def _connect_to_existing_server(self, path: Path) -> None:
+        config = get_config()
+        dragon_config_log = path / config.dragon_log_filename
+
+        if dragon_config_log.is_file():
+
+            dragon_confs = self._parse_launched_dragon_server_info_from_files(
+                [dragon_config_log]
+            )
+            logger.debug(dragon_confs)
+            for dragon_conf in dragon_confs:
+                if not "address" in dragon_conf:
```

Comment:
Could we put validation into the parsing code? e.g. `if not "address" in ...` could just filter out the conf returned from `_parse_launched_dragon_server_info_from_files` and we sort of trust "the things the parser returns are valid".